src/dialer/queueManager.js

```diff
@@ -54,10 +62,18 @@ class DialQueueManager {
       return
     }
 
-    // Add the id to the general queue set if the queue isn't running
-    // and if the queue is allowed to dial
-    if (!targetQueue.isRunning && targetQueue.isDialAllowed()) {
-      this._queue.add(targetQueue.id)
+    // If dialing is not allowed, abort
+    if (!targetQueue.isDialAllowed()) {
+      return
+    }
+
+    // Add the id to its respective queue set if the queue isn't running
+    if (!targetQueue.isRunning) {
```

Comment:
Should it be possible for a dial to be added to both queues? If not we may need to check here that it's not already in the other queue.

Reply:
We should avoid them being in both. I'll update the logic to check that.

Reply:
Added some checks and tests to ensure it's only in 1 queue